src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -259,18 +259,22 @@ namespace winrt::Microsoft::Terminal::Control::implementation
     //   connection, then this'll revoke the existing connection's handlers.
     void ControlCore::Connection(const TerminalConnection::ITerminalConnection& newConnection)
     {
-        if (_connection)
+        const bool replacing = _connection != nullptr;
+        if (replacing)
         {
             _connectionOutputEventRevoker.revoke();
-
-            // Fire off a connection state changed notification, to let our hosting
-            // app know that we're in a different state now.
-            if (newConnection.State() != _connection.State())
+        }
+        // Bail early if we're not actually giving the Terminal a new connection
+        if (newConnection == nullptr)
+        {
+            if (replacing)
             {
+                _connection.Close();

```

Comment:
This isn't our job here - I know that we Start() it later because we want to maintain consistency (so that the first output gets written to the right place), but if I detach a connection _and hold a reference to it_ i would expect it to keep running.

The dtor can handle it for us if TC was holding the last reference.

Reply:
actually, maybe we should move all Start() lifecycle ownership up into the component that GIVES us a connection.

Do we have a race on Initialized() vs Connection Start?